pkg/controller/logstorage/initializer/conditions_controller.go

```diff
@@ -90,18 +91,138 @@ func (r *LogStorageConditions) Reconcile(ctx context.Context, request reconcile.
 }
+
+// resetLastTransitionTimeStamp sets the LastTransitionTime to the previously computed condition
+// when the condition status remains unchanged
+func resetLastTransitionTimeStamp(aggTigeraStatusConditions []operatorv1.TigeraStatusCondition, ls []metav1.Condition) {
+
+	for i := range aggTigeraStatusConditions {
+		for _, lsCondition := range ls {
+			if aggTigeraStatusConditions[i].Type == operatorv1.ComponentAvailable && lsCondition.Type == string(operatorv1.ComponentReady) ||
+				aggTigeraStatusConditions[i].Type == operatorv1.ComponentDegraded && lsCondition.Type == string(operatorv1.ComponentDegraded) ||
+				aggTigeraStatusConditions[i].Type == operatorv1.ComponentProgressing && lsCondition.Type == string(operatorv1.ComponentProgressing) {
+				// Retain the old LastTransitionTime when the condition type's status is unchanged.
+				// TODO remove comment - Compare the messages in case when progressing or degraded is still true with one or more component flicker between degraded and progressing.
+				if string(aggTigeraStatusConditions[i].Status) == string(lsCondition.Status) && aggTigeraStatusConditions[i].Message == lsCondition.Message {
+					aggTigeraStatusConditions[i].LastTransitionTime = lsCondition.LastTransitionTime
+				}
+			}
+		}
+	}
+}
+
+func setAndClearTigeraStatus(aggTigeraStatusConditions []operatorv1.TigeraStatusCondition, conditionType operatorv1.StatusConditionType, clearTypes []operatorv1.StatusConditionType, reason, message string) {
+
+	for i := range aggTigeraStatusConditions {
+		if aggTigeraStatusConditions[i].Type == conditionType {
+			aggTigeraStatusConditions[i].Status = operatorv1.ConditionTrue
+			aggTigeraStatusConditions[i].Message = fmt.Sprintf("%s:%s", message, strings.TrimSuffix(aggTigeraStatusConditions[i].Message, ","))
+		} else {
+			for _, clearType := range clearTypes {
+				if aggTigeraStatusConditions[i].Type == clearType {
+					aggTigeraStatusConditions[i].Status = operatorv1.ConditionFalse
+					aggTigeraStatusConditions[i].Message = ""
+				}
+			}
+		}
+		aggTigeraStatusConditions[i].Reason = reason
+		if conditionType == operatorv1.ComponentAvailable {
+			aggTigeraStatusConditions[i].Message = message
+		}
+	}
+}
+
+func updateAggregatedConditions(tsConditions []operatorv1.TigeraStatusCondition, aggTigeraStatusConditions []operatorv1.TigeraStatusCondition, statusMap map[operatorv1.StatusConditionType]bool, logStorageType string) {
+	for _, condition := range tsConditions {
+		for i := range aggTigeraStatusConditions {
+			if aggTigeraStatusConditions[i].Type == condition.Type {
+
+				statusMap[condition.Type] = statusMap[condition.Type] || (condition.Status == operatorv1.ConditionTrue)
+				// Available should be marked true when all of the subcontrollers are available.
+				if aggTigeraStatusConditions[i].Type == operatorv1.ComponentAvailable {
+					statusMap[condition.Type] = statusMap[condition.Type] && (condition.Status == operatorv1.ConditionTrue)
+				}
```

Comment:
Add this additional check since i noticed. LogStorage User had tigerastatus with all conditions set to False.
`log-storage-users     False       False         False       6m58s`